tools/runner/logsaver.go

```diff
@@ -0,0 +1,110 @@
+package runner
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	grpcv1 "github.com/grpc/test-infra/api/v1"
+	clientset "github.com/grpc/test-infra/clientset"
+	"github.com/grpc/test-infra/status"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LogSaver provides functionality to save pod logs to files.
+type LogSaver struct {
+	clientset clientset.GRPCTestClientset
```

Comment:
This might becomre podsGetter corev1.PodsGetter

Reply:
Oh yes, I really like the idea of giving the LogSaver only what it needs